rtrlib/pfx/trie/trie-pfx.c

```diff
@@ -336,7 +337,14 @@ inline void pfx_table_free_reason(struct pfx_record **reason, unsigned int *reas
         *reason_len = 0;
 }
 
-int pfx_table_validate_r(struct pfx_table *pfx_table, struct pfx_record **reason, unsigned int *reason_len, const uint32_t asn, const struct lrtr_ip_addr *prefix, const uint8_t prefix_len, enum pfxv_state *result)
+RTRLIB_EXPORT int pfx_table_validate_r(
```

Comment:
Should we also export their variants? 
- `pfx_table_validate`
- `pfx_table_for_each_rec`

Reply:
I don't think so.
The only reason I export `pfx_table_validate_r` is because there is nothing providing it's functionality on the mgr level and I don't see a use case for `pfx_table_for_each_rec`, as the only thing it has over the `for_each_ip*_record` functions is the ability to iterate a subtree, which does not make sense because nothing publicly visible returns a `trie_node`.

Reply:
Maybe it's not much effort to provide a `pfx_table_validate_r` function on the rtrmgr level instead.

Reply:
sure, but I would not be comfortable with removing `pfx_table_validate_r` from the export table anyway. Because I believe the chances somebody actually uses this functions are rather high.

Reply:
@mroethke `pfx_table_validate_r` is used by `rpki-rov.c` for example.